graphql/batch_executor.go

```diff
@@ -195,6 +195,9 @@ func executeNonBatchWorkUnit(ctx context.Context, src interface{}, dest *outputN
 // returns new work units that are required to resolve the rest of the
 // query result.
 func resolveBatch(ctx context.Context, sources []interface{}, typ Type, selectionSet *SelectionSet, destinations []*outputNode) ([]*WorkUnit, error) {
+	if len(sources) == 0 {
+		return nil, nil
```

Comment:
please add a unit test

Reply:
Added a test that fails without this fix.